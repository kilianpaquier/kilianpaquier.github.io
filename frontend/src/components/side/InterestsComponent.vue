<template>
    <v-row :no-gutters="true">
        <v-col cols="12">
            <h2 class="mb-3">{{ $t("projects_title") }}</h2>
        </v-col>
        <v-col cols="12">
            <!-- Interests -->
            <v-row justify="center">
                <v-col v-for="(interest, index) in interests" :key="'interest_' + index" class="text-center" cols="auto">
                    <div>
                        <v-img :class="borderColor()" :eager="true" :cover="true" :src="interest.img" class="rounded-circle mx-auto mb-3 interests-img w-100 h-100"></v-img>
                        <span class="body-2">{{ $t(interest.title) }}</span>
                    </div>
                </v-col>
            </v-row>
        </v-col>
    </v-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {useTheme} from "vuetify";

export default defineComponent({
    name: "InterestsComponent",

    data: () => ({
        interests: [
            {
                title: "interests[0]",
                img: "/interests/basketball.png",
            },
            {
                title: "interests[1]",
                img: "/interests/movies.png",
            },
            {
                title: "interests[2]",
                img: "/interests/videogames.png",
            },
        ]
    }),

    setup: () => {
        const theme = useTheme();

        return {
            borderColor: () => theme.global.current.value.dark ? "interests-border-dark" : "interests-border"
        }
    }
});
</script>

<style lang="scss" scoped>
.interests-border {
  box-shadow: 0 0 0 7px #eeeeee;
  -moz-box-shadow: 0 0 0 7px #eeeeee;
  -o-box-shadow: 0 0 0 7px #eeeeee;
  -webkit-box-shadow: 0 0 0 7px #eeeeee;
}

.interests-border-dark {
  box-shadow: 0 0 0 7px #424242;
  -moz-box-shadow: 0 0 0 7px #424242;
  -o-box-shadow: 0 0 0 7px #424242;
  -webkit-box-shadow: 0 0 0 7px #424242;
}

.interests-img {
  min-height: 90px;
  min-width: 90px;
  max-width: 90px;
  max-height: 90px;
}
</style>