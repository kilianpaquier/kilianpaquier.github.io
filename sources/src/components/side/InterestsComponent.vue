<template>
  <v-container fluid>
    <!-- Interests title -->
    <h2
      class="font-weight-bold mb-3"
      v-text="$vuetify.lang.t(`$vuetify.interests-title`)"
    />

    <!-- Interests imgs -->
    <v-row justify="center">
      <v-col
        cols="auto"
        class="text-center px-5"
        v-for="(interest, i) in interests"
        :key="i"
      >
        <rounded-outlined-img :object="interest" />
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import Vue from "vue";
import { Interest } from "@/model/entities";
import Component from "vue-class-component";
import RoundedOutlinedImg from "@/components/common/RoundedOutlinedImg.vue";

@Component({
    components: { RoundedOutlinedImg },
})
export default class InterestsComponent extends Vue {
  interests: Interest[] = [
      {
          title: "interests.basketball",
          img: "basketball.png",
      },
      {
          title: "interests.tvtime",
          img: "movies.png",
      },
      {
          title: "interests.games",
          img: "videogames.png",
      },
  ];
}
</script>
