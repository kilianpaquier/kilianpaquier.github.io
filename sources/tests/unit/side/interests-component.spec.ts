import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import InterestsComponent from "@/components/side/InterestsComponent.vue";
import Vuetify from "vuetify";
import Vue from "vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("InterestsComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(InterestsComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create InterestsComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
