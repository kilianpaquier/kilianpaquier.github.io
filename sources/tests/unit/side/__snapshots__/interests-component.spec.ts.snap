// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestsComponent.vue should create InterestsComponent 1`] = `
<div class="container container--fluid">
  <h2 class="font-weight-bold mb-3">$vuetify.interests-title</h2>
  <div class="row justify-center">
    <div class="text-center px-5 col col-auto">
      <div>
        <div class="v-image v-responsive rounded-circle mx-auto mb-3 interests-border interests-img theme--light">
          <div class="v-responsive__content"></div>
        </div> <span class="body-2">$vuetify.interests.basketball</span>
      </div>
    </div>
    <div class="text-center px-5 col col-auto">
      <div>
        <div class="v-image v-responsive rounded-circle mx-auto mb-3 interests-border interests-img theme--light">
          <div class="v-responsive__content"></div>
        </div> <span class="body-2">$vuetify.interests.tvtime</span>
      </div>
    </div>
    <div class="text-center px-5 col col-auto">
      <div>
        <div class="v-image v-responsive rounded-circle mx-auto mb-3 interests-border interests-img theme--light">
          <div class="v-responsive__content"></div>
        </div> <span class="body-2">$vuetify.interests.games</span>
      </div>
    </div>
  </div>
</div>
`;
